docs/codecs.rst

```diff
@@ -1,11 +1,207 @@
+            }
+        },
+    }
+
+    
+Format and algorithm
+~~~~~~~~~~~~~~~~~~~~
+
+Encoding and decoding is performed using the algorithm defined in
+[RFC1951]_.
+
+Encoded data should conform to the Gzip file format [RFC1952]_.
+
+
+Blosc
+-----
+
+Codec URI:
+    https://purl.org/zarr/spec/codecs/blosc
+
+    
+Configuration parameters
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+cname:
+    A string identifying the internal compression algorithm to be
+    used. At the time of writing, the following values are supported
+    by the c-blosc library: "lz4", "lz4hc", "blosclz", "zstd",
+    "snappy", "zlib".
+    
+clevel:
+    An integer between 1 and 9 indicating the compression level.
+
+shuffle:
+    An integer value in the set {0, 1, 2, -1}. A value of 1
+    indicates that byte-wise shuffling is performed in addition to
+    compression. A value of 2 indicates the bit-wise shuffling is
+    performed in addition to compression. If a value of -1 is given,
+    then default shuffling is used: bit-wise shuffling for buffers
+    with item size of 1 byte, byte-wise shuffling otherwise.
+    Shuffling is turned off completely when the value is 0.
+
+blocksize:
+    An integer giving the size in bytes of blocks into which a
+    buffer is divided before compression.
+
+For example, the array metadata document below specifies that the
+compressor is the Blosc codec configured with a compression level of
+1, byte-wise shuffling, the ``lz4`` compression algorithm and the
+default block size::
+
+    {
+        "compressor": {
+            "codec": "https://purl.org/zarr/spec/codecs/blosc",
+            "configuration": {
+                "cname": "lz4",
+                "clevel": 1,
+                "shuffle": 1,
+                "blocksize": 0
+            }
+        },
+    }
+
+
+Format and algorithm
+~~~~~~~~~~~~~~~~~~~~
+
+Blosc is a meta-compressor, which divides an input buffer into blocks,
+then applies an internal compression algorithm to each block, then
+packs the encoded blocks together into a single output buffer with a
+header. The format of the encoded buffer is defined in [BLOSC]_. The
+reference implementation is provided by the `c-blosc library
+<https://github.com/Blosc/c-blosc>`_.
+
+
+Deprecated codecs
+=================
+
+There are no deprecated codecs at this time.
+
+
+References
+==========
+
+.. [RFC2119] S. Bradner. Key words for use in RFCs to Indicate
+   Requirement Levels. March 1997. Best Current Practice. URL:
+   https://tools.ietf.org/html/rfc2119
+
+.. [RFC1951] P. Deutsch. DEFLATE Compressed Data Format Specification version
+   1.3. Requirement Levels. May 1996. Informational. URL:
+   https://tools.ietf.org/html/rfc1951
+
+.. [RFC1952] P. Deutsch. GZIP file format specification version 4.3.
+   Requirement Levels. May 1996. Informational. URL:
+   https://tools.ietf.org/html/rfc1952
+
+.. [BLOSC] F. Alted. Blosc Chunk Format. URL:
+   https://github.com/Blosc/c-blosc/blob/master/README_CHUNK_FORMAT.rst
```

Comment:
This document describes the "Blosc Chunk Format". Does it mean a Zarr chunk consists of one or more Blosc chunks? If so, is the Zarr chunk a simple concatenation of the Blosc chunks?

Reply:
The "Blosc Chunk Format" describes how blosc encodes an input buffer. So one zarr chunk becomes one blosc chunk once encoded.

Reply:
:+1: